lib/taste_tester/commands.rb

```diff
@@ -210,5 +211,187 @@ def self.upload
+      # may have different tags or labels assigned to the "master" branch,
+      # (i.e. 'master', 'stable', etc.) and should be configured if different
+      # than the defaults.
+      start_ref = ''
+      type = repo.class.name.split('::').last
+      case type
+      when 'Svn'
+        start_ref = repo.latest_revision
+      when 'Git'
+        start_ref = TasteTester::Config.vcs_master_git
+      when 'Hg'
+        start_ref = TasteTester::Config.vcs_master_hg
+      end
+
+      changeset = BetweenMeals::Changeset.new(
+        logger,
+        repo,
+        start_ref,
+        nil, # compare with current working copy
+        {
+          :cookbook_dirs =>
+            TasteTester::Config.relative_cookbook_dirs,
+          :role_dir =>
+            TasteTester::Config.relative_role_dir,
+        },
+        @track_symlinks,
+      )
+
+      return changeset
+    end
+
+    def self._find_impact(changeset)
+      if TasteTester::Config.relative_cookbook_dirs.length > 1
+        logger.error('Knife deps does not support multiple cookbook paths.')
+        logger.error('Flatten the repository or use the custom_impact hook.')
+        exit(1)
+      end
+
+      cookbooks = Set.new(changeset.cookbooks)
+      roles = Set.new(changeset.roles)
+
+      if cookbooks.empty? && roles.empty?
+        logger.error('No cookbooks or roles have been modified.')
+        exit(1)
+      end
+
+      unless cookbooks.empty?
+        logger.warn('Modified Cookbooks:')
+        cookbooks.each { |cb| logger.warn("\t#{cb}") }
+      end
+      unless roles.empty?
+        logger.warn('Modified Roles:')
+        roles.each { |r| logger.warn("\t#{r}") }
+      end
+
+      return _find_impact_roles(cookbooks, roles)
+    end
+
+    def self._find_impact_roles(cookbooks, roles)
+      # Now that we have a list of changed cookbooks and roles, we can check
+      # which roles contain modified dependencies. For simplicity, we will
+      # check using brute force by iterating through each role and comparing
+      # its dependencies with the set of modified files. If a match exists,
+      # the role was impacted, otherwise we can move on.
+      # This may change in the future if it is a perfomance bottleneck.
+
+      role_dir = TasteTester::Config.role_dir
+      chef_dir = File.join(TasteTester::Config.repo,
+                           TasteTester::Config.base_dir)
+
+      # CLI options for knife call
+      config = "--config #{TasteTester::Config.knife_config}"
+      chef_path = "--chef-repo-path #{chef_dir}"
+      options = '--tree --recurse'
+
+      # shell out to knife once and parse the resulting file for dependents
+      # if knife did not exit with 0, print whatever it returned and exit
+      logger.warn('Finding dependencies (this may take a minute or two)...')
+      knife = Mixlib::ShellOut.new(
+        "knife deps /#{role_dir}/*.rb #{options} #{config} #{chef_path}",
+      )
+      knife.run_command
+      if knife.error?
```

Comment:
This whole if statement is just:

```
knife.error!('knife deps failed, please check output below')
```

Reply:
ShellOut#error! does not take any arguments, so I would have to go with the default `Expected process to exit with #{valid_exit_codes.inspect}, but received '#{exitstatus}'`
This is probably ok though since my message wasn't much more helpful.